packages/qdrant-loader-core/tests/test_logging_core.py

```
@@ -61,6 +61,203 @@ def emit(self, record):
+    Starlette/FastAPI percent-decode query keys during parsing, so
+    "sec%72et=<value>" is accepted as "secret=<value>" by the app while the
+    raw request line uvicorn logs keeps it percent-encoded. A literal-match
+    filter would miss it and leak the secret in plaintext.
+    """
+    logging_mod = import_module("qdrant_loader_core.logging")
+    UvicornAccessRedactFilter = logging_mod.UvicornAccessRedactFilter
+
+    captured_messages = []
+
+    class TestHandler(logging.Handler):
+        def emit(self, record):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `emit`

Add return type annotation: `None`

(ANN202)